nitro-node/package.json

```diff
@@ -0,0 +1,62 @@
+{
+  "name": "@janhq/nitro-node",
+  "version": "1.0.0",
+  "description": "This NodeJS library is a wrapper for Nitro, a lightweight (3mb) inference engine written in C++. See nitro.jan.ai",
+  "main": "dist/index.js",
+  "node": "dist/index.cjs.js",
+  "author": "Jan <service@jan.ai>",
+  "license": "AGPL-3.0",
+  "scripts": {
+    "test": "jest --verbose --detectOpenHandles",
+    "pretest": "node download-nitro.js",
+    "build": "tsc --module commonjs && rollup -c rollup.config.ts",
+    "downloadnitro": "node download-nitro.js",
+    "build:publish": "npm pack",
+    "postinstall": "node postinstall.js"
```

Comment:
What would happen if there were a module that depend on this module? e.g. module wrapper such as nitro-extension. 
1. Will post-install work with "npm install" command?
2. Will post-install step runs as soon as user install wrapper module?

Reply:
I packed `nitro-node` as a tarball and then install from that and it works. But will need to test for some more cases. Let me setup CI later to verify this, if not then I'll expose the function to let wrapper download the binaries.

Reply:
Please see the fourth commit here: https://github.com/janhq/jan/pull/1712/commits
I installed from git subdirectory using gitpkg and the binaries are downloaded in `node_modules/@janhq/nitro-node` although in the source there is no `bin`.

Reply:
> I packed `nitro-node` as a tarball and then install from that and it works. But will need to test for some more cases. Let me setup CI later to verify this, if not then I'll expose the function to let wrapper download the binaries.

I mean, you packaged as a tarball on MacOS computer, there would be MacOS bin files, but when you put the Tarball in Windows, that would break.

Reply:
Please test with both NPM & Yarn, scenario:
User might just add nitro-node into their package dependencies, it should work without any additional steps (only npm / yarn install)

Reply:
Ah, I understand your point, the tarball should be generic so there is no `bin` in it also, as it is not defined in the `files` properties of `package.json`. Let me continue the verification of `postinstall` hook when I make the github action workflow for `nitro-node` then.

Reply:
Please see [this comment](https://github.com/janhq/nitro/pull/367#issuecomment-1911079924) for the test results regarding this discussion.